microcloud/cmd/microcloud/main_init.go

```diff
@@ -414,6 +421,76 @@ func AddPeers(sh *service.Handler, systems map[string]InitSystem) error {
 	return nil
 }
 
+// checkClustered checks whether any of the selected systems have already initialized a service.
+func checkClustered(s *service.Handler, autoSetup bool, serviceType types.ServiceType, systems map[string]InitSystem) (string, map[string]string, error) {
+	var firstInitializedSystem string
+	var existingMembers map[string]string
+
+	// We don't allow appending to an existing LXD cluster.
+	if serviceType == types.LXD {
+		return "", nil, nil
```

Comment:
When MicroCloud first starts, it already checks whether LXD is initialized before it restarts LXD [here](https://github.com/canonical/microcloud/blob/81d501c885e4fd559842b2654714672f667d55d0/microcloud/cmd/microcloud/main_init.go#L89) to ensure the correct symlinks to MicroOVN and MicroCeph are present. We do this so we don't accidentally restart LXD while it's running instances for example.

This condition is just a convenience that lets us exit early from the `checkClustered` function in the case of LXD because it will always be uninitialized if we got to this point. 

I thought it might keep the invocation simple since we can just iterate over every `serviceType` and call `checkClustered` for all of them and get a sane value. The alternative would be that we need to remember to always call `checkClustered` in an if block where `serviceType != types.LXD`.

Reply:
>  because it will always be uninitialized if we got to this point.

I was not thinking about this, thanks, clear now! I think it would be helpful to extend the comment with the reason why an empty string is returned here.

Reply:
Updated the comment and labelled the args so it should be a bit clearer now. Don't know why the preview is still showing the old version.